auto_inertial_params/proposal.md

```diff
@@ -0,0 +1,410 @@
+the 6 are needed since the matrix is symmetric. 
+
+>**Note:** If the axis about which the MOI Tensor is calculated, is taken to be the 
+principal axis of inertia, then the products of inertia become 0 and the matrix becomes a diagonal matrix.
+
+All these values of MOI Tensor can be calculated as follows:
+
+$$\begin{eqnarray} 
+ I\_{11} = I\_{xx} = \int (y^2 + z^2)dm \\\
+ I\_{22} = I\_{yy} = \int (x^2 + z^2)dm \\\
+ I\_{33} = I\_{zz} = \int (x^2 + y^2)dm \\\
+\end{eqnarray}$$
+
+Here \\(dm\\) is the mass of an infinitesimal unit of the object and \\(x, y, z\\) are the distances of that unit from axes. 
+
+Similarly, the Products of Inertia can be calculated as:
+
+$$\begin{eqnarray}
+ I\_{12}  = I\_{xy} = \int -xydm = I\_{yx} = I\_{21} \\\
+ I\_{13}  = I\_{xz} = \int -xzdm = I\_{zx} = I\_{31} \\\
+ I\_{23}  = I\_{yz} = \int -yzdm = I\_{zy} = I\_{32} \\\
+\end{eqnarray}$$
+
+In this solution, the **infinitesimal element** of the object can be **represented by the each Voxel.** 
+Instead of calculating the mass \\(dm\\) for each voxel, we will calculate the volume \\(dv\\) of each 
+voxel using the voxel size (the simple volume formula of a cube can be used because voxels are cubes). 
+Then mass for each element would be mass density, \\(\rho\\) multiplied by the volume, \\(dv\\). 
+Considering the mass density to be constant and substituting in the above equations we get:
+
+$$\begin{eqnarray}
+I\_{11} = I\_{xx} = \rho\int (y^2 + z^2)dv \\\
+I\_{22} = I\_{yy} = \rho\int (x^2 + z^2)dv \\\
+I\_{33} = I\_{zz} = \rho\int (x^2 + y^2)dv \\\
+I\_{12}  = I\_{xy} = \rho\int -xydv = I\_{yx} = I\_{21} \\\
+I\_{13}  = I\_{xz} = \rho\int -xzdv = I\_{zx} = I\_{31} \\\
+I\_{23}  = I\_{yz} = \rho\int -yzdv = I\_{zy} = I\_{32} \\\
+\end{eqnarray}$$
+
+#### Advantages of the Voxelization Approach
+ * Voxelization is a generalized approach that would work for all kinds 
+ of meshes (convex and non-convex meshes).
+ * It can be made configurable by allowing the user to set an appropriate `voxel_size` 
+ for the process. A smaller `voxel_size` would be comparitively computationally heavy 
+ but would provide closer to real world value for moment of inertia.
+ * Voxelization is more intuitive way of moment of inertia calculations as 
+ compared to other integral methods.
+
+### Integration-based numerical method
+It uses Gauss’s Theorem and Greene’s Theorem of integration to convert volume 
+integrals to surface integrals (Gauss’s Theorem) and then surface integrals to line 
+integrals(Greene’s Theorem)<sup>[\[5\]](#References)</sup>.
+This method works for triangle meshes which are simple water-tight polyhedrons. 
+Currently, the origin of the mesh being used needs to be set at the geometric centre 
+to obtain the correct value.
+Since this method uses the vertex data for calculations, a high vertex count would be 
+required for near-ideal values. For eg, in case of a cylinder, it was observed that with 
```

Comment:
This is important for usage right? Should this be documented some place else as well? Probably a Limitations or Considerations sub-section in implementation?

Reply:
The Proposed Implementation section of the proposal focuses more on the implementation from libsdformat's side. Therefore, instead of adding it there, I created a sub-section in the `Integration-based Numerical Method` and added this point. I have added some other points as well that explain this method's limitations and suitable use cases. The changes can be found in [fca06ae](https://github.com/gazebosim/sdf_tutorials/pull/92/commits/fca06ae64a6843311fa125ff89c6f9a376c37599)

Reply:
Great thanks!